res_syntax/tests/ppx/react/expected/firstClassModules.res.txt

```diff
@@ -57,7 +57,10 @@ module Select = {
     type key
     type t
   }
-  type props<'model, 'selected, 'onChange, 'items> = {
+  type key
+  and a
```

Comment:
This declares a new type a in the current scope. Will error if a type called a is already in scope. Eg defined just above.
The (type a) should be confined to the body of make.

Reply:
Yes, I think so too. I want to avoid to add type declaration. But I don't have a clear idea to fix this:
```rescript
module type T = {
  type t
}

module V4A2 = {
  type props<'foo> = {foo: 'foo}

  let make = (type a, {foo, _}: props<'foo>) => {
    module T = unpack(foo: T with type t = a)
    foo
  }
  let make = {
    let \"Newtype$V4A2" = (props: props<_>) => make(props)

    \"Newtype$V4A2"
  }
}
```

Reply:
If it compiles OK, should be good.
Might be worth testing a couple of examples by hand and see that it compiles OK, before changing the ppx.

Reply:
That `: props<'foo>` should be `: props<_>` I think.

Reply:
> That `: props<'foo>` should be `: props<_>` I think.

Yes, It works. But How can JSX4 know that foo is going be unpack in the body expression?

```rescript
// original code
module V4A2 = {
  @react.component
  let make = (type a, ~foo) => {
    module T = unpack(foo: T with type t = a)
    foo
  }
}
```

Reply:
The current ppx passes the core_type of argument to props type. Do you mean passing no parameter to props type?
```rescript
@react.component
let make = (~a: int) => React.int(a)

// (current) transformed to
type props<'a> = {a: 'a}
let make = ({a, _}: props<int>) => React.int(a)
                         ^^^^^

// (new?)
type props<'a> = {a: 'a}
let make = ({a, _}: props<_>) => React.int(a)
                         ^^^^^
```

Reply:
Ah no that should stay the same.
I think above you want `let make = (type a, ...   : props<a>`